### PR TITLE
refactor: Refactored `test_assertions.py` file.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description 
Refactored `test_assertions.py` file. Used `with` when opening files to ensure closure, as opening files in this way is more concise, and also ensures that the call to `file.close()` is not skipped if there is an exception. Also, removed unnecessary calls to `.keys()` when iterating over a dictionary as the default behaviour when iterating over a dictionary is to iterate over the keys.

## Checklist 
- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### Socials: 
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

**Comment by Mr-Niraj-Kulkarni** [comment]
@Sai-Suraj-27, Can you please tag the issue for which these changes were required to be made. Thanks

**Comment by Sai-Suraj-27** [comment]
> @Sai-Suraj-27, Can you please tag the issue for which these changes were required to be made. Thanks

Sir, I made these on my own to make the code better and clean. This is a refactoring PR. It's not related to any issue.

File: ivy_tests/test_ivy/test_misc/test_assertions.py [hunk 1]
```diff
@@ -40,16 +40,14 @@
...
-    sys.stdout = f
-    lines = ""
-    try:
-        check_all(results)
-    except Exception as e:
-        print(e)
-    sys.stdout = orig_stdout
-    f.close()
-
+    with open(filename, "w") as f:
```
Comment (vedpatwardhan) [inline ivy_tests/test_ivy/test_misc/test_assertions.py:43]: let's add a new line before and after this block similar to how it's done with the next blocks in this function. Same applies to the other changes made

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @Sai-Suraj-27, just requested a minor change, rest looks good to merge, thanks 😄(thanks for requesting the review @Mr-Niraj-Kulkarni)

File: ivy_tests/test_ivy/test_misc/test_assertions.py (hunk 1 above)
Comment (Sai-Suraj-27) [reply ivy_tests/test_ivy/test_misc/test_assertions.py:44]: @vedpatwardhan sir, I made the suggested changes, Please look into it now, Thank you.

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge the PR @Mr-Niraj-Kulkarni, thanks @Sai-Suraj-27 😄

**Pull request merged**